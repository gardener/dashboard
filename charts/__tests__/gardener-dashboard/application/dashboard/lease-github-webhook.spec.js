//
// SPDX-FileCopyrightText: 2023 SAP SE or an SAP affiliate company and Gardener contributors
//
// SPDX-License-Identifier: Apache-2.0
//

'use strict'

const { helm } = fixtures

const renderTemplates = helm.renderDashboardApplicationTemplates

describe('gardener-dashboard', function () {
  describe('lease-github-webhook', function () {
    let templates

    beforeEach(() => {
      templates = [
        'lease-github-webhook'
      ]
    })

    it('should not render the template with default values', async function () {
      const values = {}
      const documents = await renderTemplates(templates, values)
      expect(documents).toHaveLength(1)
      const [leaseGithubWebhook] = documents
      expect(leaseGithubWebhook).toMatchSnapshot()
    })

    it('should render the template if gitHub webhook is configured', async function () {
      const values = {
        global: {
          dashboard: {
            gitHub: {
              webhookSecret: 'foobar'
            }
          }
        }
      }
      const documents = await renderTemplates(templates, values)
      expect(documents).toHaveLength(1)
      const [leaseGithubWebhook] = documents
      expect(leaseGithubWebhook).toMatchSnapshot()
    })
  })
})
