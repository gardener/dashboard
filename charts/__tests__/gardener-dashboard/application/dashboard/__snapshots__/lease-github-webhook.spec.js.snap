// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gardener-dashboard lease-github-webhook should not render the template with default values 1`] = `null`;

exports[`gardener-dashboard lease-github-webhook should render the template if gitHub webhook is configured 1`] = `
Object {
  "apiVersion": "coordination.k8s.io/v1",
  "kind": "Lease",
  "metadata": Object {
    "labels": Object {
      "app.kubernetes.io/component": "dashboard",
      "app.kubernetes.io/instance": "gardener-dashboard",
      "app.kubernetes.io/managed-by": "Helm",
      "app.kubernetes.io/name": "gardener-dashboard",
      "helm.sh/chart": "gardener-dashboard-application-0.1.0",
    },
    "name": "gardener-dashboard-github-webhook",
    "namespace": "garden",
  },
  "spec": Object {
    "holderIdentity": "",
    "renewTime": "1970-01-01T00:00:00.000000Z",
  },
}
`;
